src/os/bluestore/Allocator.h

```diff
@@ -55,6 +55,7 @@ class Allocator {
   virtual void dump() = 0;
   virtual void foreach(
     std::function<void(uint64_t offset, uint64_t length)> notify) = 0;
+  virtual const uint64_t* get_as_bitmap(size_t* out_count) const { return nullptr; }
```

Comment:
An open question:
is it a good practice to add a function to interface, in cases where we know that only one instance will be able to supply implementation.
Is not dynamic_cast better way to address this type of specific extensions?

Reply:
IMO neither is ideal.  But I'm hesitant to expose BitmapAllocator to BlueStore. On the other hand if we do such an exposure then we can implement all the mechanics without dynamic_cast usage which is a bit better.